excellent/evaluator_test.go

```diff
@@ -85,7 +85,7 @@ func TestEvaluateTemplate(t *testing.T) {
 		{"@array1d", array1d},
 		{"@array1d.0", xs("a")},
 		{"@array1d.1", xs("b")},
-		{"@array2d.0.2", xs("c")},
+		{"@array2d.0.2", ERROR}, // need to use square brackets
```

Comment:
Previously `@(array2d.0.2)` was an error because the parser considers `0.2` to be a valid expression, but `@array2d.0.2` would parse because it's just splitting on `.`. Now they both go thru the parser.

Reply:
Maybe should add the `@array2d.0["2"]` as a test case?

Reply:
`foo[123]` vs `foo["arst"]` vs `foo[foo.bar]` -> `foo.arst`, `foo.2auth` needs to be `foo["2auth"]`

Reply:
does `foo[0]` match `{"0": "hi hi"}`?